apps/files_sharing/lib/Controller/ShareAPIController.php

```diff
@@ -536,8 +536,8 @@ public function deleteShare(string $id): DataResponse {
 	 * @param string|null $path Path of the share
 	 * @param int|null $permissions Permissions for the share
 	 * @param int $shareType Type of the share
-	 * @param string|null $shareWith The entity this should be shared with
-	 * @param string $publicUpload If public uploading is allowed
+	 * @param ?string $shareWith The entity this should be shared with
+	 * @param ?string $publicUpload If public uploading is allowed
```

Comment:
```suggestion
	 * @param 'true'|'false'|null $publicUpload If public uploading is allowed, deprecated (explain how to do without)
```

If I understand correctly this is deprecated? ("legacy")

Reply:
Yes it is not used anymore since we have the new sharing flow (28? 27?).
It is pretty useless and just enforces some permissions (READ for false and READ+MODIFY+UPLOAD+DELETE for true).

We do the permissions manually  nowadays

Reply:
```suggestion
	 * @param 'true'|'false'|null $publicUpload If public uploading is allowed
```

Reply:
Please mark the parameter as deprecated then.